### ISSUE TITLE
CMD does not show Unix LF when pasting

### ISSUE THREAD

**Issue by Biswa96** [microsoft/Terminal#663]
* Build 18890 (18894 on the way...)
* What doing: Create a file with **Unix (LF) line endings**. For example,

```cmd
echo foo
echo bar
```
Paste it in CMD window. The output is `fooecho bar`. The output should be in separate lines. Try the same in mintty. It shows correct output.

**Comment by zadjii-msft** [comment]
I think this is a notoriously long-standing bug with cmd.exe. @paulcam206 might have more to share on this topic. I'm fairly sure this falls under the category of "we can't change cmd.exe anymore".

**Comment by Biswa96** [comment]
Oooo... You remembered me miniksa's PTSD story. Powershell works correctly. I assume the issue may be on CMD not ConHost or OpenConsole :)

**Comment by FrankHB** [comment]
I agree this should better be fixed, but it can also be a feature :)

I'm working on a REPL which accepts one line per input by default. The language is free-form, which can have multiple lines in the script files. I save some pieces of test code in the script files, and each test case can have more than one lines of code. The LF ending makes more than one lines can be selected together and copied. When they are pasted into the console, the REPL ignores the LF characters, so these lines are automatically concatenated into one. If the bug is fixed, such trick/abuse will be broken.

**Comment by Biswa96** [comment]
Any 'official' reply?

**Comment by zadjii-msft** [comment]
I don't know what's at the root of this problem, if it's a recent regression or not, or if it's truly a bug or just "weird things that conhost does that we can't fix cause legacy reasons". I'm throwing it on the 20H1 milestone to make sure someone investigates, though it might be a while. If someone else is passionate about fixing this and would like to dig into _why_ this is happening, power to you :)

**Comment by Biswa96** [comment]
> fixing this and would like to dig into why this is happening

cmd.exe is not open source.

**Comment by zadjii-msft** [comment]
Yea but conhost is, and conhost is going to be _far_ more valuable in debugging this than cmd.exe.

**Comment by zadjii-msft** [comment]
At first I thought: "this is something that we can't fix because it's legacy cmd madness". But now I'm not so sure.

Lets look at what happens when you try to paste 
```cmd
echo foo
echo bar
echo baz
```

with only LF line endings:
![image](https://user-images.githubusercontent.com/18356694/65890203-1d593400-e368-11e9-9cbe-340eb883d160.png)


This makes it seem like `COOKED_READ` in conhost is not properly breaking up the lines on a LF-only paste. In fact, the problem might be even higher, in our reading from the clipboard itself. Let's look at this code, which is responsible for taking the clipboard data and generating `INPUT_RECORD`s from it:
https://github.com/microsoft/terminal/blob/083be43700c01604983a9ff67e9fe6edab8ad0e5/src/interactivity/win32/Clipboard.cpp#L140-L183

The most interesting bit is on line 171:

``` c++
     if (IsInVirtualTerminalInputMode() && currentChar == UNICODE_LINEFEED) 
     {
           currentChar = UNICODE_CARRIAGERETURN; 
     } 
```

When we're in VT _input_ mode (read:wsl), we'll turn single LF characters into a CR before handling the paste. This was in order to solve microsoft/WSL#2006.

If I take out the `IsInVirtualTerminalInputMode` check, this will fix the cmd scenario:
![image](https://user-images.githubusercontent.com/18356694/65891140-aa50bd00-e369-11e9-99b7-0a708a22b1be.png)

Now the real question is: Is this a safe change? What will this possibly regress? Let's discuss! @miniksa @DHowett-MSFT @carlos-zamora 

<hr>

Also for linking internally: MSFT:20169262 deals with this for batch files, which is almost certainly unsolvable.
